src/App.tsx

```diff
@@ -1,61 +1,131 @@
+import { useState } from 'react';
 import './App.scss';
+// import cn from 'classnames';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TasksWithTodo } from './types/Todo';
+import { TodoList } from './components/TodoList';
+
+function getUserById(userId: number) {
+  return usersFromServer.find(user => user.id === userId) || null;
+}
+
+const tasksWithTodo: TasksWithTodo[] = todosFromServer.map(todo => ({
+  ...todo,
+  user: getUserById(todo.userId),
+}));
 
 export const App = () => {
+  const [title, setTitle] = useState('');
+  const [currentUserId, setCurrentUserId] = useState(0);
+  const [todos, setTodos] = useState(tasksWithTodo);
+  const [hasTitleError, setHasTitleError] = useState(false);
+  const [hasUserError, setHasUserError] = useState(false);
+  const getTodoId = tasksWithTodo.map(todo => todo.id);
+  const getId = Math.max(...getTodoId) + 1;
+
+  const handleTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newTitle = event.target.value;
+
+    setTitle(newTitle);
+
+    if (newTitle) {
+      setHasTitleError(false);
+    }
+  };
+
+  const handleUserChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    const newUser = +event.target.value;
+
+    setCurrentUserId(newUser);
+
+    if (newUser !== 0) {
+      setHasUserError(false);
+    }
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+
+    if (!title) {
+      setHasTitleError(true);
+    } else {
+      setHasTitleError(false);
+    }
+
+    if (!currentUserId) {
+      setHasUserError(true);
+    } else {
+      setHasUserError(false);
+    }
```

Comment:
```suggestion
      setHasUserError(!!currentUserId);
```

also here can be more simplify